### PR TITLE
Handle blocks that can not be loaded.

### PR DESCRIPTION
**Pull request by hanslovsky**
Currently, blocks that can not be loaded (`N5Reader.readBlock` returns `null`) defaults to leaving the corresponding `Img` unchanged, i.e. it will hold the default value for the appropriate primitive type. As discussed in #2 it can be desirable to set the corresponding `Img` to a different value, or throw an exception. This PR 
 - adds a `Consumer< Img >` member to the `N5CellLoader` that let's the user specify how to handle unavailable blocks. The default behavior remains unchanged.
 - adds appropriate convenience methods to `N5Utils`

### CODE REVIEW

File: src/main/java/org/janelia/saalfeldlab/n5/imglib2/N5Utils.java [hunk 1]
```diff
@@ -323,16 +325,118 @@ private static void cropBlockDimensions(
...
+	{
+		return openSparseWithDiskCache( n5, dataset, N5CellLoader.setToDefaultValue( defaultValue ) );
+	}
+
+	/**
+	 * Open an N5 dataset as a memory cached {@link LazyCellImg}.
+	 *
+	 * @param n5
+	 * @param dataset
+	 * @param ioExceptionHandler
```
Comment (axtimwalde) [inline src/main/java/org/janelia/saalfeldlab/n5/imglib2/N5Utils.java]: name discrepancy

File: src/main/java/org/janelia/saalfeldlab/n5/imglib2/N5CellLoader.java [hunk 2]
```diff
@@ -158,4 +192,24 @@ public void load( final SingleCellArrayImg< T, ? > cell )
 			throw new IllegalArgumentException( "Type " + dataType.name() + " not supported!" );
 		}
 	}
+
+	/**
+	 *
+	 * @return {@link Consumer} that does not do anything.
+	 */
+	public static < T > Consumer< T > noOpConsumer()
```
Comment (axtimwalde) [inline src/main/java/org/janelia/saalfeldlab/n5/imglib2/N5CellLoader.java]: Remove and replace `N5CellLoader.noOpConsumer()` calls by `t -> {}` which is a lot shorter and readable or re-use same instance, i.e.
```java
private Consumer<T> noOpConsumer = t ->{};
public static <T> Consumer<T> noOpConsumer() {
  return noOpConsumer;
}

File: src/main/java/org/janelia/saalfeldlab/n5/imglib2/N5Utils.java [hunk 3]
```diff
@@ -323,16 +325,118 @@ private static void cropBlockDimensions(
...
+	 */
+	public static final < T extends NativeType< T > > RandomAccessibleInterval< T > openVolatile(
+			final N5Reader n5,
+			final String dataset ) throws IOException
+	{
+		return openSparseVolatile( n5, dataset, N5CellLoader.noOpConsumer() );
+	}
+
+	/**
+	 * Open an N5 dataset as a disk-cached LazyCellImg.  Note that this
```
Comment (axtimwalde) [inline src/main/java/org/janelia/saalfeldlab/n5/imglib2/N5Utils.java]: {@link LazyCellImg}

File: src/main/java/org/janelia/saalfeldlab/n5/imglib2/N5Utils.java [hunk 4]
```diff
@@ -34,6 +35,7 @@
 import net.imglib2.cache.img.DiskCachedCellImgOptions;
 import net.imglib2.cache.img.LoadedCellCacheLoader;
 import net.imglib2.cache.ref.SoftRefLoaderCache;
+import net.imglib2.img.Img;
```
Comment (axtimwalde) [inline src/main/java/org/janelia/saalfeldlab/n5/imglib2/N5Utils.java]: Where does that come from?

File: src/main/java/org/janelia/saalfeldlab/n5/imglib2/N5Utils.java [hunk 5]
```diff
@@ -323,16 +325,118 @@ private static void cropBlockDimensions(
...
+	public static final < T extends NativeType< T > > RandomAccessibleInterval< T > openSparseVolatile(
+			final N5Reader n5,
+			final String dataset,
+			final T defaultValue ) throws IOException
+	{
+		return openSparseVolatile( n5, dataset, N5CellLoader.setToDefaultValue( defaultValue ) );
+	}
+
+	/**
+	 * Open an N5 dataset as a disk-cached LazyCellImg.  Note that this
```
Comment (axtimwalde) [inline src/main/java/org/janelia/saalfeldlab/n5/imglib2/N5Utils.java]: {@link LazyCellImg}

File: src/main/java/org/janelia/saalfeldlab/n5/imglib2/N5Utils.java [hunk 6]
```diff
@@ -414,19 +518,21 @@ private static void cropBlockDimensions(
 	 *
 	 * @param n5
 	 * @param dataset
+	 * @param ioExceptionHandler
```
Comment (axtimwalde) [inline src/main/java/org/janelia/saalfeldlab/n5/imglib2/N5Utils.java]: parameter name

File: src/main/java/org/janelia/saalfeldlab/n5/imglib2/N5Utils.java [hunk 7]
```diff
@@ -505,24 +611,26 @@ private static void cropBlockDimensions(
 
 	/**
 	 * Open an N5 dataset as a disk-cached LazyCellImg.  Note that this
```
Comment (axtimwalde) [inline src/main/java/org/janelia/saalfeldlab/n5/imglib2/N5Utils.java]: {@link LazyCellImg}

File: src/main/java/org/janelia/saalfeldlab/n5/imglib2/N5Utils.java [hunk 8]
```diff
@@ -505,24 +611,26 @@ private static void cropBlockDimensions(
...
-	 * requires that al part of the the N5 dataset that will be accessed fit
+	 * requires that all parts of the the N5 dataset that will be accessed fit
 	 * into /tmp.
 	 *
 	 * @param n5
 	 * @param dataset
+	 * @param ioExceptionHandler
```
Comment (axtimwalde) [inline src/main/java/org/janelia/saalfeldlab/n5/imglib2/N5Utils.java]: parameter name

File: src/main/java/org/janelia/saalfeldlab/n5/imglib2/N5Utils.java (hunk 4 above)
Comment (hanslovsky) [reply src/main/java/org/janelia/saalfeldlab/n5/imglib2/N5Utils.java]: [Here](https://github.com/saalfeldlab/n5-imglib2/pull/3/files/aae06c3187ba808a542545a3e4e83356ae4fb062#diff-35426c6c65fe6d01148b0db841c6b851R433)

**Comment by hanslovsky** [comment]
Addressed the comments in 3b99782

This PR should be squashable into master (I just recently noticed that github offers squashing through the web interface).

Is there an eclipse formatter scheme for N5? I am using the imglib2-clean-up formatter (as a save action) and it introduced a lot of irrelevant formatting changes, and figuring out what part of the diff is relevant is an unnecessary waste of time.

**Comment by axtimwalde** [comment]
Good point.  n5-imglib2 used imglib2-clean-up, but it shouldn't.  I delayed reformatting until after the next meaningful merge.  Saalfeld lab projects should use this style (that I finally stored somewhere) :

https://github.com/saalfeldlab/n5/blob/master/doc/n5-eclipse-style.xml

It's Java conventions with fixed tabs and some reduced spacing.

With modern fonts, I find the ImgLib2 style too space consuming.

**Comment by axtimwalde** [comment]
Thanks!

**Comment by axtimwalde** [comment]
(of course I just had to update the style...)

**Comment by hanslovsky** [comment]
Resolved with #3, this can be closed now.

**Pull request merged**